enhancements/network/ovn-on-hypershift.md

```diff
@@ -0,0 +1,271 @@
+
+**ovsdb-server**: *should* not require any changes. This depends on Raft working when nodes are identified by DNS (and IPs change).
+
+**OVN-Kubernetes:** *may* need some changes to perform leader election via the management cluster. Otherwise, it doesn't need to be aware of the management cluster.
+
+#### OVN DBs
+
+The OVN databases (nbdb, sbdb) will continue to use the internal raft-based replication. However, they will be deployed as a StatefulSet instead. Rather than using HostPath volumes, they will use whatever persistent storage volumes are available in the management cluster.
+
+#### Node-DB connectivity
+By (ab)using the TLS Passthrough feature of OpenShift Routes, we can expose the southbound database directly to the client cluster.
+
+### Monitoring
+
+The control plane components, as pods in the management cluster, will need to be monitored by the management monitoring infrastructure. We will need to create a ServiceMonitor for each control plane in question.
+
+One complication is that we should create AlertManager alerts on a per-customer basis. That means, inevitably, that we will need some kind of management layer for these alerts. What form this takes is not yet defined.
+
+### Risks and Mitigations
+
+**OVSBD Raft and DNS** - the ovsdb team does not currently test Raft with dynamic member IPs. While the team believes this should work, it must be added to their test plan. Also, we have missed the window for any new changes to OVS for the 4.11 release. If any changes to ovsdb are needed, we either need exceptional processes or workarounds.
+
+### Open Questions 
+
+
+What storage / volume types will be available for the OVN DBs to use? Can we assume a functioning CSI environment? Will this storage have sufficient performance? Can we generate a reasonable size prediction? To answer this, we will need to utilize the existing Perf&Scale processes to characterize performance.
+
+What bandwidth will the sbdb traffic consume? Will the added latency cause any issues? Should we enable ovn-controller conditional monitoring? At the time this document was being drafted, the OVN team is re-examining the performance impacts of ovn-controller's conditional monitor feature.
+
+If the bandwidth or performance is not acceptable, will we have to deploy some kind of ovsdb proxy on the infra nodes? Assuming we cannot *require* infra nodes, will we nevertheless need to make that option available to customers?
+
+How do we report control-plane status -- which references "invisible" pods -- to the end-user? Does the end-user need to know that ovsdb is somehow degraded? What do we show in the `network` ClusterOperator object?
```

Comment:
Metrics should report up to the control plane and will be shipped off to RHOBS, at least in the SD use case.  We're drawing the line at the control plane except for what is needed to support HPA / VPA, customers don't get the control plane metrics.

Reply:
They have no visibility into the db at all, so it's probably not worth reporting.  The customer visible side effects would be things like pod creation latency, or in the worst case, the inability to start up new pods if the network control plane is locked up completely.

Can an admin of a hosted cluster run `oc get clusteroperators` ?  What will they see?  Will they see any of the things running on the management cluster?  I'd guess they wouldn't see it at all?  I need to get hypershift running again and check I guess ...

Reply:
Ok, I got on a hosted cluster and confirmed you see all the cluster operators

```
oc get clusteroperators
NAME                                       VERSION       AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
console                                    4.10.0-rc.1   True        False         False      44h     
csi-snapshot-controller                    4.10.0-rc.1   True        False         False      4d15h   
dns                                        4.10.0-rc.1   True        False         False      4d15h   
image-registry                             4.10.0-rc.1   True        False         False      4d15h   
ingress                                    4.10.0-rc.1   True        False         False      4d15h   
kube-apiserver                             4.10.0-rc.1   True        False         False      4d15h   
kube-controller-manager                    4.10.0-rc.1   True        False         False      4d15h   
kube-scheduler                             4.10.0-rc.1   True        False         False      4d15h   
kube-storage-version-migrator              4.10.0-rc.1   True        False         False      4d15h   
monitoring                                 4.10.0-rc.1   True        False         False      4d15h   
network                                    4.10.0-rc.1   True        False         False      4d15h   
node-tuning                                4.10.0-rc.1   True        False         False      4d15h   
openshift-apiserver                        4.10.0-rc.1   True        False         False      4d15h   
openshift-controller-manager               4.10.0-rc.1   True        False         False      4d15h   
openshift-samples                          4.10.0-rc.1   True        False         False      4d15h   
operator-lifecycle-manager                 4.10.0-rc.1   True        False         False      4d15h   
operator-lifecycle-manager-catalog         4.10.0-rc.1   True        False         False      4d15h   
operator-lifecycle-manager-packageserver   4.10.0-rc.1   True        False         False      4d15h   
service-ca                                 4.10.0-rc.1   True        False         False      4d15h   
storage                                    4.10.0-rc.1   True        False         False      4d15h   
```

So this info is still visible, but you're right that less is actionable by the owner of the hosted cluster. 

@derekwaynecarr @csrwng Do you have any guidance for any changes we should make to how the cluster-network-operator reports its status for hosted clusters?  Or just leave it the same and info that isn't helpful to the target cluster owner may still be helpful if they have to report it back to whoever is running the management cluster?

Reply:
I'd say we can leave it as is now. If necessary we can reconcile the status of the HostedCluster to reflect the status reported inside the guest cluster's cno clusteroperator resource.

Reply:
My first plan is to just say something like

- "Control plane upgrade is rolling out, 1 out of 3 pods updated"
- "Control plane is degraded: ovn-db has 1 unready pod"

and discard any pod-specific information.